test/test_nn.py

```diff
@@ -6682,69 +6685,69 @@ def perm_fn(x):
         self.assertEqual(tuple(result.shape), tuple(ref_output.shape))
```

Comment:
Hm I think something got messed up in this file- shouldn't the logic be indented underneath the `batch_first` for loop?

Reply:
Hmm...earlier this morning there was a conflict when merging, so I accepted all incoming changes in VSCode. I did not pay much attention and just saw `torch.Tensor` -> `torch.tensor`. Then I found there are things missing. You want to me to revert back an earlier version of `test_nn.py` that passed the test?

Reply:
Sure, reverting sounds good! As you mentioned, something landed while this was in review to change all references of `torch.Tensor` -> `torch.tensor` in `test_nn.py`. Alongside that, constant tensors were changed to floats (e.g. `10` -> `10.`). This should be straightforward albeit tedious to fix in the reverted `test_nn.py`.

Reply:
Now it should be done. Last night I was trying to do a revert but GitKraken says the action will be destructive when I tried to push, so this morning I did a manual copy-pasta. I verified the newly modified test here: 
https://github.com/scaomath/torch-playground/blob/1537ec97ade78de4f64b528cfa349acf024510e7/test_transformer.py#L1-L17
Runs no problem locally.